### DISCUSSION TITLE
storeToRefs breaks reactivity with Ref<Set>

### DISCUSSION THREAD

**Discussion by ryanolf-tb** [vuejs/pinia#2754] [Help and Questions] [answered]
### Reproduction

https://play.pinia.vuejs.org/#eNqNlU1v2kAQhv/KyBeMCnaiVKmEAJFWkdoe2ipwKz0Ye0w28X5odw1IyP+9s+uPGIJoT9gzz8zOvjMejsGDUtGuxGASTE2qmbJg0JYKikRsZ+vAmnUwXwvGldQWjlAaXFqpESrIteQwiOJUlsKijqwZ9EDjqJV8wtx0rGKCJT1on9j0eQTWUS1DtRCxFqkUhhBD7gpm/XRhW0M4HLacdxP25mo9acHS16+JyArUBIRDmM3huBZQx0RJlq3kMqxfBIWdOFaNg5OjcmX5kkOq6l0i8eEDMcQ1iDfaCyBvwWlcK0760otFrorEIr0BTDeltVLAwpdPfehfgzrykGUkGwggC/BpXON1qJqLCXTpYDfOpaYMouQgc3D9PB7ppapGMI27U6exasP5lXD7j3B3q95NglFQN3vMExW9GClo0LwW68ZB9UxqdZzNT4izrIPYP0do+Hij5d7QhL0QPGrRBU1KnOHOSlmYcaJYHfZsrTKTOE4zQXyGBdvpSKCNheLxu5jFfXQf3cYF28R0UMxEhofTYyhinCH/n+wturiJbj9Gn3zW3V2TlLusLimNUUWyWEPzmbPtmSip5IpRj38qy2h+T8RJikLuv3ub1SV2JabPmL5esL+YQ131L40k3w5717KJ3iI107kflz/wQM+dk8usLIi+4nxCI4vS1Vhjn8tmNDvOV/vN95iJ7co8HiwK017KFerV8LyX+cuVq7+Vexfd9VR82z0kYbd7MsyZONtT57sHNOads106ePDOenN0q27WTxgOmjMHJ592HSKIpbzhjd8jtY1fsJnGJnAPS7RT+p42qOfh7z9upbWUvUo5Li9F6sSCZouxCdTQsJXORLukKP02C5nPTcqdB64uBNpLge5H09+DFiSgGAEfgVvQo+b85nflmkMhQfUXRSwjWw==

### Steps to reproduce the bug

Click on the "add to n and m" button.

### Expected behavior

The sets of `n` and `m` values should grow.

### Actual behavior

Only the set of `m` grows.

The values of each `n` and `m` are added to sets `s` and `t`, respectively. A watcher is meant to increment `n` and `m` when the respective set is added to. However, the watcher of the ref obtained from storeToRefs does not trigger.

**Comment by ryanolf-tb** [comment]
Maybe it's a separate issue, but in the example you can also see that assigning a `new Set` to the `store.t` breaks the reactive connection to the watcher. My understanding from https://masteringpinia.com/blog/ref-vs-reactive-in-stores was that this is also not the expected behavior.

**Comment by posva** [comment, marked as answer]
The problem lies on the `watch(s, ...` Since s is a ref of the set, it will only trigger if the value changes, that means if a new set is assigned. For what you want, [you need to use `deep: true` in your watcher](https://play.pinia.vuejs.org/#eNqNVU1v2kAQ/SsjXzAq2IlSpRICRFpFantoq8Ct9GDsMdnE+6HdtUFC/PfOrj9iCKI92Tvz3uzMm/H4EDwoFVUlBpNgalLNlAWDtlRQJGI7WwfWrIP5WjCupLZwgNLg0kqNcIRcSw6DKE5lKSzqyJpBD2gcaiWfMDcdVjHBkh5ol9j0eQTWoVoM5UKItUilMAQx5D7CrB8ubHMIh8MW590Ee3O1nrRg6evXRGQFagKEQ5jN4bAWUHOiJMtWchnWB0G0E8eqcXByHF1aPuWQsnoXSHz4QJgRlZ4hqglYXZJKxGsoHmQvELknEnAa1x0gvelgkasisUgngOmmtFYKWPhyqC/9sqhDD1lGMoIAsgCfxjW8pqq5mEAXDqpxLjVFECUHmYPr7+FAhyPlPo27W6exaun8Ct3+g+6q6lUSjIK6+WOeqOjFSEGD57VYNw7KZ1Kr42x+YpxlHcT+PULDxxstd4Ym7oXAoxa6oMmJM6yslIUZJ4rVtGdrlZnEcZoJwmdYsEpHAm0sFI/fcRb30X10GxdsE9NFMRMZ7k+vIcY4Q/4/0Vvo4ia6/Rh98lGruyYod1FdUBqrI8liDc1rzrZnoqSSK0Y9/qkso3k+EScpCrn77m1u2LoU02dMXy/YX8y+zvqXRpKvwl5ZNtFbpGY69+PyB+7pvXNymZUFoa84n9DIonQ51rDPZTOaHc5n+833mIntyjzuLQrTFuUS9Wp4vJf5y5XS39K9i+56Kr7tIpKw20UZ5kyc7a3zXQQa887ZLiHce2e9SbrVN+sHDAfNnYOTT7umCMJS3PDG75Xaxi/YTGMTuIMl2il9TxvU8/D3H7fiWpS9inK4vBSpEwuarcYmUIOGrXQmqpKi9NstZD42KXdOXF0g2ktE99D0u9CCBBQj4CNwC3vU3N88V645RAmOfwFK9yiX).

**Comment by ryanolf-tb** [comment]
Thanks for the quick solution. I had thought `watch` was deeply reactive for `ref` but apparently that's not true with `Set`. Thanks again.
